### PR TITLE
fs/mmap: fix mmap returned address

### PR DESCRIPTION
**Pull request by XuNeo**
## Summary

- 1. rammap should return memory start address
- 2. update output `mapped` pointer  in file_mmap_ before returning

## Impact

No.

## Testing

mmap used in gettext now works correctly.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@jlaitine could you review the change? we found some regression from https://github.com/apache/nuttx/pull/8026

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jlaitine — APPROVED** [review]
LGTM, but there is another issue left, maybe look into it at the same?

File: fs/mmap/fs_rammap.c [hunk 1]
```diff
@@ -235,7 +237,6 @@ int rammap(FAR struct file *filep, FAR struct mm_map_entry_s *entry,
 
   /* Add the buffer to the list of regions */
 
-  entry->vaddr = rdbuffer;
   entry->priv.i = kernel;
   entry->munmap = unmap_rammap;
```
Comment (jlaitine) [inline fs/mmap/fs_rammap.c:242]: the next ret check with goto errout_with_region is actually wrong as well; the pointer to rdbuffer is no longer pointing to the allocated area, but is still freed there

Comment (jlaitine) [reply fs/mmap/fs_rammap.c:242]: In the "free" below we should use entry->vaddr, since that stores the original allocated ptr

**Comment by jlaitine** [comment]
Clarified my original comment, it was too quickly written to be understandable....

**Comment by XuNeo** [comment]
Let me update the commit later.

For `int rammap(FAR struct file *filep, FAR struct mm_map_entry_s *entry, bool kernel)`, am I correct to add comment to say that `entry->offset` and `entry->length` must be initialized ?

File: fs/mmap/fs_rammap.c (hunk 1 above)
Comment (XuNeo) [reply fs/mmap/fs_rammap.c:241]: Done.

**Review by jlaitine — APPROVED** [review]
Great, thanks!

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore the unrelated ci broken which is fixed by https://github.com/apache/nuttx/pull/8155.

**Pull request merged**